### ISSUE TITLE
BlockUI: widgetVar is reset to null after partial updates

### ISSUE THREAD

**Issue by Rasoul26** [primefaces/primefaces#8494]
I am using Primefaces 11, and I am adding this components in home.xhtml page:

```html
<p:dataTable id="dataTable" var="n" value="#{homeBean.dataMedel}" lazy="true" paginator="true" rows="10"
             paginatorTemplate="{CurrentPageReport} {FirstPageLink} {PreviousPageLink} {NextPageLink} {RowsPerPageDropdown}" 
             sortOrder="descending" rowsPerPageTemplate="10,15,20,30,50">
```
```html
<p:blockUI widgetVar="bui" block="dataTable" animate="false">
     <p:graphicImage value="ajax_loader.gif"/>
</p:blockUI>
```
```html
<h:selectOneMenu id="filterTopic" value="#{filterBean.topic}" label="topic" title="topic"
    <p:ajax event="change" update="dataTable" onstart="PF('bui').show();" oncomplete="PF('bui').hide()"
            listener="#{homeBean.onPreRender}"/>
    <f:selectItems value="#{topicBean.topicList}"/>
</h:selectOneMenu>
```

when I select one of the items in filterTopic menu, the ajax_loader.gif should fall on the dataTable until the data is fetch from homeBean. but it`s not working and i can see in the console the following error:

**Widget for var 'bui' not available!**  -->   core.js.xhtml?ln=primefaces&v=11.0.0:18

**home.xhtml Uncaught TypeError: Cannot read properties of null (reading 'hide')**  -->  home.xhtml

also when refresh page and run PrimeFaces.widgets in browser console, result is:

     bui: c {_super: undefined, cfg: {…}, id: 'form:j_id_bo', jqId: '#form\\:j_id_bo', jq: S.fn.init(1), …}

but when select one of the items in filterTopic menu, bui value set to null:

     bui: null

also this code worked with primefaces 10

**Comment by tandraschko** [comment]
if you provide a primefaces-test project, we can check it. Otherwise its invalid.

**Comment by mohsen** [comment]
Let me put it simple in plain English.

We have a form with a `p:datatable` and a `p:blockUI` (with widgetvar=**bui**) inside form as datatable sibling. With versions 8/10 or earlier `p:blockUI` and its widgetvar are working and valid after partial `p:datatable` updates. BUT in PF 11 after any such updates, `PrimeFaces.widgets['bui']` is reset to `null`, so all widgetvar actions on the blockUI throws exception.

**Comment by tandraschko** [comment]
as i said, if you provide a reproducer (forked primefaces-test), we can have a look

**Comment by melloware** [comment]
Please  provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Comment by melloware** [comment]
@Rasoul26 closing this issue until a runnable reproducer is provided so we can debug.

**Comment by melloware** [comment]
cc @jepsar

**Comment by Rasoul26** [comment]
Hello

This issue was resolved by changing `block="dataTable"` to `block="@form"` for `p:blockUI` tag in primefaces 8 & 11.

Thanks for your effort

**Comment by melloware** [comment]
OK thanks for posting back.  @jepsar is seeing a similar issue so I am wondering if its truly and issue we should try and fix.

**Comment by jepsar** [comment]
Reproducer:

```xhtml
<h:form id="frmTest">
    <h:panelGroup id="panel">
      Panel
    </h:panelGroup>
    <p:blockUI widgetVar="blockPanel" block="panel" blocked="true"/>
    <p:commandButton value="Unblock"
                     update="panel"
                     oncomplete="PF('blockPanel').hide()"/>
</h:form>
```

**Comment by melloware** [comment]
I wonder if this was broken on PF 7.0, or 8.0.  Might have to look through BlockUI commits.

**Comment by jepsar** [comment]
It's removed here:
https://github.com/primefaces/primefaces/blob/e57b27fa6b7893a2a75a3c4d71be13b80867c760/primefaces/src/main/resources/META-INF/resources/primefaces/core/core.ajax.js#L1280-L1304

**Comment by melloware** [comment]
So isn't this line wrong too..
```js
PrimeFaces.widgets[widgetVar] = null;
```

That just makes it NULL but its still in the list as undefined.  Should it remove it from the widgets array?

**Comment by melloware** [comment]
it should do.
```js
delete PrimeFaces.widgets[widgetVar];
```

**Comment by melloware** [comment]
Reproducer:
[pf-8494.zip](https://github.com/primefaces/primefaces/files/8218119/pf-8494.zip)

**Comment by melloware** [comment]
OK i see why its happening its because the BlockUI panel `jq` object gets created inside the panel. So when the panel is removed from the DOM so it he BlockUI Mask which causes the widget to be marked for removal.

**Comment by melloware** [comment]
I think #5866 is related and was never solved

**Comment by tandraschko** [comment]
i will close this in favor of: https://github.com/primefaces/primefaces/issues/8548

it needs a bit more refactoring how the widget works
